pkg/provider/aws/resources/aws_resource_value.go

```diff
@@ -25,9 +31,52 @@ func (r *AwsResourceValue) SetResource(res core.Resource) {
 }
 
 func (val *AwsResourceValue) UnmarshalYAML(value *yaml.Node) error {
+	type intermediate struct {
+		Type     string
+		Resource map[interface{}]interface{}
+		Property string
+	}
+	in := intermediate{}
+	err := value.Decode(&in)
+	if err != nil {
+		return err
+	}
+	val.PropertyVal = in.Property
+	if in.Type == "" {
+		return nil
+	}
+
+	typeToResource := make(map[string]core.Resource)
+	for _, res := range ListAll() {
+		typeToResource[res.Id().Type] = res
+	}
+	// Subnets are special because they have a type that is not the same as their resource type since it uses a characteristic of the subnet
+	typeToResource["subnet_private"] = &Subnet{}
+	typeToResource["subnet_public"] = &Subnet{}
```

Comment:
This feels like it should be done in `ListAll`

Reply:
yeah i think you may be right, probably easier to do that